sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -974,3 +974,13 @@ case object UnresolvedWithinGroup extends LeafExpression with Unevaluable {
   override def dataType: DataType = throw new UnresolvedException("dataType")
   override lazy val resolved = false
 }
+
+case class UnresolvedTranspose(
+    indexColumn: Option[Expression],
```

Comment:
shall we make it a `indices: Seq[Expression]`?
currently we only support
1, `len(indices) == 0` => no specified index;
2, `len(indices) == 1` => a specified index;
by assert(len(indices)) == 0 or 1.

Then this parameter could support multi-level indices if needed in the future

Reply:
SGTM, the protobuf message needs to be stable and it's better to use `Seq[Expression]` here to be future proof.

Reply:
Makes sense! Adjusted both node and protobuf message.